sdk/core/Azure.Core/src/ResourceIdentifier.cs

```diff
@@ -389,10 +444,11 @@ public override int GetHashCode()
         /// <param name="resourceName"> The name of the resource.</param>
         /// <returns> The combined resource id. </returns>
         [EditorBrowsable(EditorBrowsableState.Never)]
+#pragma warning disable CA1822
         public ResourceIdentifier AppendProviderResource(string providerNamespace, string resourceType, string resourceName)
         {
             ValidateProviderResourceParameters(providerNamespace, resourceType, resourceName);
-            return new ResourceIdentifier(this, providerNamespace, resourceType, resourceName);
+            return new ResourceIdentifier($"{_stringValue}/providers/{providerNamespace}/{resourceType}/{resourceName}");
```

Comment:
Does this call a constructor that parses the full string again?  What is the thinking behind this change?

Reply:
Now that ctor is lazy we don't parse it again immediately.

The thinking is that we'll need the full string value anyway and other properties are lazy. This also simplifies the logic a lot because the source of truth is the same in all cases.

In the future, if we find a scenario where transferring parsed values benefits the performance we can copy them here.

Reply:
This will make this much slower in the hot path since we use each of the keys inside the resourceidentifier to construct the uri for API calls

Reply:
My understanding is that was temporary and you have an issue to move to using the full RID.

Reply:
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/network/Azure.ResourceManager.Network/src/Generated/Subnet.cs#L109

Reply:
As an example here is the ctor + accessors for the hotpath with new code

|           Method |     Mean |     Error |    StdDev |  Gen 0 | Allocated |
|----------------- |---------:|----------:|----------:|-------:|----------:|
| CreateResourceId | 1.271 us | 0.0151 us | 0.0142 us | 0.3948 |      2 KB |

Here is the ctor + accessors for the hotpath synced to d732dc4

|           Method |     Mean |   Error |  StdDev |  Gen 0 |  Gen 1 | Allocated |
|----------------- |---------:|--------:|--------:|-------:|-------:|----------:|
| CreateResourceId | 490.2 ns | 1.92 ns | 1.60 ns | 0.1831 | 0.0010 |      1 KB |

```
    private const string IdString = "/subscriptions/0c2f6471-1bf0-4dda-aec3-cb9272f09575/resourceGroups/myRg/providers/Microsoft.Compute/location/uswest/virtualMachines/myVm";

    [Benchmark]
    public ResourceIdentifier CreateResourceId()
    {
        ResourceIdentifier id = new ResourceIdentifier(IdString);
        string x = id.Name;
        x = id.ResourceGroupName;
        x = id.SubscriptionId;
        x = id.Parent.Name;
        x = id.Parent.Parent.Name;
        return id;
    }
```

Reply:
Reverter.